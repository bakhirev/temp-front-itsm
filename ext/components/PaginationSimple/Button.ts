import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../common';

const SIZE = 40;

interface IButtonProps {
  selected?: boolean;
  disabled?: boolean;
  withEllipsis?: boolean;
}

export const Button = styled.button<IButtonProps>`
  ${({ disabled, selected, theme, withEllipsis }) => css`
    align-items: center;
    background: ${selected ? theme.color.primary[60] : 'transparent'};
    border-radius: ${theme.borderRadius};
    border: none;
    color: ${selected ? theme.color.neutral.white : theme.color.neutral[90]};
    cursor: ${withEllipsis ? 'default' : 'pointer'};
    display: flex;
    height: ${SIZE}px;
    justify-content: center;
    outline: none;
    user-select: none;
    width: ${SIZE}px;

    & svg {
      fill: ${theme.color.neutral[90]};
    }

    &:focus {
      background: ${disabled || selected || withEllipsis ? '' : theme.color.opacity.blackFocus};
    }

    &:hover {
      background: ${disabled || selected || withEllipsis ? '' : theme.color.opacity.blackHover};
    }

    &:active {
      background: ${disabled || selected || withEllipsis ? '' : theme.color.opacity.blackPressed};
    }

    &:disabled {
      background: transparent;
      color: ${theme.color.neutral[30]};
      cursor: default;

      & svg {
        fill: ${theme.color.neutral[30]};
      }
    }
  `};
`;

Button.defaultProps = {
  theme: DEFAULT_THEME,
};
