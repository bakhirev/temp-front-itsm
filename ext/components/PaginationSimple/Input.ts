import styled, { css } from 'styled-components';

import { TYPOGRAPHY } from '../Typography';
import { DEFAULT_THEME } from '../common';

export const Input = styled.input`
  ${({ theme }) => css`
    background: transparent;
    border-radius: ${theme.borderRadius};
    border: 2px solid transparent;
    box-sizing: border-box;
    color: ${theme.color.neutral[90]};
    font-family: ${TYPOGRAPHY.fontFamily};
    font-size: ${TYPOGRAPHY.fontSize[14]};
    font-weight: ${TYPOGRAPHY.fontWeight.regular};
    height: 40px;
    line-height: ${TYPOGRAPHY.lineHeight[20]};
    margin-left: 8px;
    outline: none;
    padding-left: 14px;
    width: 114px;
    -moz-appearance: textfield;

    &::placeholder {
      color: ${theme.color.primary[60]};
      font-weight: ${TYPOGRAPHY.fontWeight.medium};
      opacity: 1;
    }

    &::-webkit-inner-spin-button,
    &::-webkit-outer-spin-button {
      -webkit-appearance: none;
    }

    &:hover {
      background: ${theme.color.opacity.blackHover};
    }

    &:focus {
      background: ${theme.color.neutral.white};
      border-color: ${theme.color.primary[60]};
    }
  `};
`;

Input.defaultProps = {
  theme: DEFAULT_THEME,
};
