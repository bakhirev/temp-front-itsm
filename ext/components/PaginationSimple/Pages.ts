import styled from 'styled-components';

const MARGIN_HORIZONTAL = 8;

export const Pages = styled.div`
  display: flex;
  height: 100%;
  margin: 0 ${MARGIN_HORIZONTAL}px;
`;
