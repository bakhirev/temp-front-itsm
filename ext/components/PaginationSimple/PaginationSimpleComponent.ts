import styled from 'styled-components';

const HEIGHT = 40;

export const PaginationSimpleComponent = styled.div`
  display: flex;
  height: ${HEIGHT}px;
  justify-content: center;
`;
