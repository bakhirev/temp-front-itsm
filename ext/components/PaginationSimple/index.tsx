import React, { useRef, useState } from 'react';
import type { ChangeEvent, FC, KeyboardEvent, MouseEvent } from 'react';

import { ReactComponent as ChevronLeftOutline } from '../Icons/system/ChevronLeftOutline.svg';
import { ReactComponent as ChevronRightOutline } from '../Icons/system/ChevronRightOutline.svg';
import { Body1Short } from '../Typography';
import { KEY_CODES } from '../common';

import { Button } from './Button';
import { Input } from './Input';
import { Pages } from './Pages';
import { PaginationSimpleComponent } from './PaginationSimpleComponent';

const BOUNDARY_PAGES = 5;
const ELLIPSIS = '...';
const FIRST_PAGE = 1;
const ICON_SIZE = 24;
const PAGES_WITHOUT_ELLIPSIS = 7;
const PAGES_WITHOUT_INPUT = 21;
const PLACEHOLDER = 'На страницу';
const SIBLING_PAGES = 2;
const SIBLING_PAGES_MOBILE = 1;

export interface IPaginationSimpleProps {
  /** Имя класса для переопределения стилей */
  className?: string;
  /** Текущая страница */
  currentPage: number;
  /** data-test-id атрибут для тестирования компонента */
  dataTestId?: string;
  /** Список задизейбленных страниц */
  disabledPages?: Array<number>;
  /** Мобильный вид компонента */
  mobile?: boolean;
  /** Список страниц */
  pages: number;
  /** Коллбэк на изменение страницы */
  onChange: (page: number) => void;
}

export const PaginationSimple: FC<IPaginationSimpleProps> = ({
  className,
  currentPage,
  dataTestId,
  disabledPages = [],
  mobile = false,
  pages,
  onChange,
}) => {
  const [inputValue, setInputValue] = useState('');

  const inputRef = useRef<HTMLInputElement | null>(null);

  const allPages = Array.from({ length: pages }, (_, idx) => idx + 1);

  const getNumericInputValue = () => {
    const numericInputValue = parseInt(inputValue);
    if (numericInputValue < FIRST_PAGE) return FIRST_PAGE;
    if (numericInputValue > pages) return pages;
    if (isNaN(numericInputValue)) return currentPage;
    return numericInputValue;
  };

  const handleInputBlur = () => {
    onChange(getNumericInputValue());
    setInputValue('');
  };

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) =>
    setInputValue(event.target.value);

  const handleInputKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
    if (
      event.keyCode === KEY_CODES.MINUS ||
      event.keyCode === KEY_CODES.NUMPAD_SUBTRACT ||
      event.keyCode === KEY_CODES.PERIOD ||
      event.keyCode === KEY_CODES.NUMPAD_DECIMAL
    ) {
      event.preventDefault();
    }
    if (event.keyCode === KEY_CODES.ENTER) inputRef.current?.blur();
  };

  const handleButtonFocusPrevent = (event: MouseEvent<HTMLButtonElement>) => event.preventDefault();

  const getPreviousPage = () => {
    const previousPages = allPages.slice(0, currentPage - 1);
    const availablePreviousPages = previousPages.filter((page) => !disabledPages.includes(page));
    return availablePreviousPages[availablePreviousPages.length - 1];
  };

  const getNextPage = () => {
    const nextPages = allPages.slice(currentPage);
    const availableNextPages = nextPages.filter((page) => !disabledPages.includes(page));
    return availableNextPages[0];
  };

  const renderPage = (page: number) => (
    <Button
      disabled={disabledPages.includes(page)}
      key={page}
      selected={currentPage === page}
      onClick={() => onChange(page)}
      onMouseDown={handleButtonFocusPrevent}
    >
      <Body1Short>{page}</Body1Short>
    </Button>
  );

  const renderPages = (pagesList: Array<number>) => pagesList.map(renderPage);

  const renderEllipsis = () => (
    <Button tabIndex={-1} withEllipsis>
      <Body1Short>{ELLIPSIS}</Body1Short>
    </Button>
  );

  const shouldRenderEllipsis = (position: 'left' | 'right') => {
    if (mobile) {
      return position === 'left'
        ? currentPage > BOUNDARY_PAGES - 1
        : currentPage < pages - BOUNDARY_PAGES + 2;
    }

    return position === 'left'
      ? currentPage > BOUNDARY_PAGES
      : currentPage < pages - BOUNDARY_PAGES + 1;
  };

  const renderPagesWithEllipsis = () => {
    const decCurrentPage = currentPage - 1;
    const incCurrentPage = currentPage + 1;

    const startBoundaryPages = allPages.slice(0, BOUNDARY_PAGES);
    const endBoundaryPages = allPages.slice(pages - BOUNDARY_PAGES);
    const commonBoudaryPages = startBoundaryPages.filter((page) => endBoundaryPages.includes(page));

    const leftSiblingPage = currentPage - (mobile ? SIBLING_PAGES_MOBILE : SIBLING_PAGES);
    const rightSiblingPage = currentPage + (mobile ? SIBLING_PAGES_MOBILE : SIBLING_PAGES);

    const middlePages = allPages.filter((page) =>
      (startBoundaryPages.includes(page) &&
        startBoundaryPages.includes(mobile ? incCurrentPage : currentPage)) ||
      (endBoundaryPages.includes(page) &&
        endBoundaryPages.includes(mobile ? decCurrentPage : currentPage))
        ? false
        : page >= leftSiblingPage && page <= rightSiblingPage
    );

    const excludeCommonBoundaryPages = (page) =>
      commonBoudaryPages.includes(currentPage) ? !commonBoudaryPages.includes(page) : true;

    const renderStart = () =>
      startBoundaryPages.includes(mobile ? incCurrentPage : currentPage)
        ? renderPages(startBoundaryPages)
        : renderPage(FIRST_PAGE);

    const renderEnd = () =>
      endBoundaryPages.includes(mobile ? decCurrentPage : currentPage)
        ? renderPages(
            mobile ? endBoundaryPages : endBoundaryPages.filter(excludeCommonBoundaryPages)
          )
        : renderPage(pages);

    return (
      <>
        {renderStart()}
        {shouldRenderEllipsis('left') && renderEllipsis()}
        {renderPages(middlePages)}
        {shouldRenderEllipsis('right') && renderEllipsis()}
        {renderEnd()}
      </>
    );
  };

  const previousPage = getPreviousPage();
  const nextPage = getNextPage();

  return (
    <PaginationSimpleComponent className={className} data-test-id={dataTestId}>
      {!mobile && (
        <Button
          disabled={!previousPage}
          onClick={() => onChange(previousPage)}
          onMouseDown={handleButtonFocusPrevent}
        >
          <ChevronLeftOutline height={ICON_SIZE} width={ICON_SIZE} />
        </Button>
      )}
      <Pages>
        {pages > PAGES_WITHOUT_ELLIPSIS ? renderPagesWithEllipsis() : renderPages(allPages)}
      </Pages>
      {!mobile && (
        <Button
          disabled={!nextPage}
          onClick={() => onChange(nextPage)}
          onMouseDown={handleButtonFocusPrevent}
        >
          <ChevronRightOutline height={ICON_SIZE} width={ICON_SIZE} />
        </Button>
      )}
      {pages > PAGES_WITHOUT_INPUT && !mobile && (
        <Input
          placeholder={PLACEHOLDER}
          ref={inputRef}
          type="number"
          value={inputValue}
          onBlur={handleInputBlur}
          onChange={handleInputChange}
          onKeyDown={handleInputKeyDown}
        />
      )}
    </PaginationSimpleComponent>
  );
};
