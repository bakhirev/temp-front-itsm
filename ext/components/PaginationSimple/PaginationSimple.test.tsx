import React from 'react';
import { mount, shallow } from 'enzyme';

import { PaginationSimple } from '../PaginationSimple';
import type { IPaginationSimpleProps } from '../PaginationSimple';

jest.mock('../common', () => ({
  DEFAULT_THEME: {
    color: {
      neutral: {
        90: '',
        white: '',
      },
      opacity: {
        blackFocus: '',
      },
      primary: {
        60: '',
      },
    },
  },
  KEY_CODES: {
    ENTER: 13,
  },
}));

describe('PaginationSimple', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render pages without ellipsis if pages quantity is 7 or less', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render pages with ellipsis if pages quantity is 8 or more', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 8,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render input if pages quantity is 22 or more', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 22,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange with page the user clicks on', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    const thirdPage = wrapper.childAt(1).childAt(2);

    thirdPage.simulate('click');
    expect(props.onChange).toBeCalledWith(3);
  });

  it('should call onChange with next page if user clicks on next page button', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    const nextPageButton = wrapper.childAt(2);

    nextPageButton.simulate('click');
    expect(props.onChange).toBeCalledWith(2);
  });

  it('should call onChange with previous page if user clicks on previous page button', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 2,
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    const previousPageButton = wrapper.childAt(0);

    previousPageButton.simulate('click');
    expect(props.onChange).toBeCalledWith(1);
  });

  it('should call onChange with next available page if user clicks on next page button, but next page is disabled', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      disabledPages: [2],
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    const nextPageButton = wrapper.childAt(2);

    nextPageButton.simulate('click');
    expect(props.onChange).toBeCalledWith(3);
  });

  it('should call onChange with previous available page if user clicks on previous page button, but previous page is disabled', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 3,
      disabledPages: [2],
      pages: 7,
      onChange: jest.fn(),
    };

    const wrapper = shallow(<PaginationSimple {...props} />);
    const previousPageButton = wrapper.childAt(0);

    previousPageButton.simulate('click');
    expect(props.onChange).toBeCalledWith(1);
  });

  it('should call onChange with input value when user blurs input', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 22,
      onChange: jest.fn(),
    };

    const wrapper = mount(<PaginationSimple {...props} />);
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: '10' } });
    wrapper.update();
    input.simulate('blur');

    expect(props.onChange).toBeCalledWith(10);
  });

  it('should call onChange with input value when user presses enter', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 22,
      onChange: jest.fn(),
    };

    const wrapper = mount(<PaginationSimple {...props} />);
    const input = wrapper.find('input');
    const inputNode: HTMLInputElement = input.getDOMNode();
    inputNode.blur = () => input.simulate('blur');

    input.simulate('change', { target: { value: '10' } });
    wrapper.update();
    input.simulate('keydown', { keyCode: 13 });

    expect(props.onChange).toBeCalledWith(10);
  });

  it('should call onChange with first page, if input value is less than 1', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 22,
      onChange: jest.fn(),
    };

    const wrapper = mount(<PaginationSimple {...props} />);
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: '0' } });
    wrapper.update();
    input.simulate('blur');

    expect(props.onChange).toBeCalledWith(1);
  });

  it('should call onChange with last page, if input value is more than pages quantity', () => {
    const props: IPaginationSimpleProps = {
      currentPage: 1,
      pages: 22,
      onChange: jest.fn(),
    };

    const wrapper = mount(<PaginationSimple {...props} />);
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: '23' } });
    wrapper.update();
    input.simulate('blur');

    expect(props.onChange).toBeCalledWith(props.pages);
  });
});
