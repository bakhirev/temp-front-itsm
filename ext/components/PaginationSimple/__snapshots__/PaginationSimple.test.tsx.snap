// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationSimple should render input if pages quantity is 22 or more 1`] = `
<styled.div>
  <styled.button
    disabled={true}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
  <styled.div>
    <styled.button
      disabled={false}
      key="1"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={true}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        1
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="2"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        2
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="3"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        3
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="4"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        4
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="5"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        5
      </styled.div>
    </styled.button>
    <styled.button
      tabIndex={-1}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
      withEllipsis={true}
    >
      <styled.div>
        ...
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="22"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        22
      </styled.div>
    </styled.button>
  </styled.div>
  <styled.button
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
  <styled.input
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="На страницу"
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
    type="number"
    value=""
  />
</styled.div>
`;

exports[`PaginationSimple should render pages with ellipsis if pages quantity is 8 or more 1`] = `
<styled.div>
  <styled.button
    disabled={true}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
  <styled.div>
    <styled.button
      disabled={false}
      key="1"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={true}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        1
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="2"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        2
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="3"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        3
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="4"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        4
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="5"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        5
      </styled.div>
    </styled.button>
    <styled.button
      tabIndex={-1}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
      withEllipsis={true}
    >
      <styled.div>
        ...
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="8"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        8
      </styled.div>
    </styled.button>
  </styled.div>
  <styled.button
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
</styled.div>
`;

exports[`PaginationSimple should render pages without ellipsis if pages quantity is 7 or less 1`] = `
<styled.div>
  <styled.button
    disabled={true}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
  <styled.div>
    <styled.button
      disabled={false}
      key="1"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={true}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        1
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="2"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        2
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="3"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        3
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="4"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        4
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="5"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        5
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="6"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        6
      </styled.div>
    </styled.button>
    <styled.button
      disabled={false}
      key="7"
      onClick={[Function]}
      onMouseDown={[Function]}
      selected={false}
      theme={
        Object {
          "color": Object {
            "neutral": Object {
              "90": "",
              "white": "",
            },
            "opacity": Object {
              "blackFocus": "",
            },
            "primary": Object {
              "60": "",
            },
          },
        }
      }
    >
      <styled.div>
        7
      </styled.div>
    </styled.button>
  </styled.div>
  <styled.button
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    theme={
      Object {
        "color": Object {
          "neutral": Object {
            "90": "",
            "white": "",
          },
          "opacity": Object {
            "blackFocus": "",
          },
          "primary": Object {
            "60": "",
          },
        },
      }
    }
  >
    <div
      height={24}
      width={24}
    />
  </styled.button>
</styled.div>
`;
